### PR TITLE
Adding public ip option for worker nodes

### PR DESCRIPTION
**Pull request by nathan-jenan-rancher**
Adding an option to EKS driver to set the value of AssociatePublicIPAddress
in the worker nodes template.  This is so users can create a private cluster
with rancher with properly configured vpc and subnets.

Also adds validation to prevent AssociatePublicIPAddress from being set to
`false` without also passing in a custom vpc and subnets.

Issue:
rancher/rancher#15419

Depends on:
https://github.com/rancher/types/pull/591

### CODE REVIEW

File: drivers/eks/eks_driver.go [hunk 1]
```diff
@@ -241,6 +247,12 @@ func (state *state) validate() error {
 		return fmt.Errorf("virtual network, subnet, and security group must all be set together")
 	}
 
+	if state.AssociateWorkerNodePublicIP != nil &&
+		!*state.AssociateWorkerNodePublicIP &&
+		(state.VirtualNetwork == "" || len(state.Subnets) == 0) {
+		return fmt.Errorf("if private cluster is set a VPC and subnets must also be provided")
```
Comment (StrongMonkey) [inline drivers/eks/eks_driver.go]: the if checks `or` condition while the error is saying `and` is needed.

Comment (StrongMonkey) [reply drivers/eks/eks_driver.go]: discussed offline

**Comment by StrongMonkey** [comment]
LGTM

File: drivers/eks/eks_driver.go (hunk 1 above)
Comment (cjellick) [reply drivers/eks/eks_driver.go]: > if private cluster is set
should be if AssociateWorkerNodePublicIP is set to false, VPC and subnets must also be provided

**Pull request merged**